File: src/DrillingTelemetry.Contracts/TelemetryReading.cs
namespace DrillingTelemetry.Contracts;

/// <summary>
/// Represents a telemetry reading exchanged between applications.
/// </summary>
public sealed class TelemetryReading
{
    /// <summary>
    /// Gets or sets the device identifier.
    /// </summary>
    public string DeviceId { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the acquisition session that owns the sequence number.
    /// </summary>
    public Guid AcquisitionSessionId { get; set; }

    /// <summary>
    /// Gets or sets the sequence number assigned by the device simulator.
    /// </summary>
    public long SequenceNumber { get; set; }

    /// <summary>
    /// Gets or sets the identifier of the well being drilled.
    /// </summary>
    public string WellId { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the identifier of the wellbore containing the tool.
    /// </summary>
    public string WellboreId { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the distance travelled along the wellbore, in metres.
    /// </summary>
    public double MeasuredDepthMetres { get; set; }

    /// <summary>
    /// Gets or sets the operation that governed measured-depth movement when
    /// the reading was acquired.
    /// </summary>
    public DrillingOperation DrillingOperation { get; set; }

    /// <summary>
    /// Gets or sets the signed measured-depth change rate, in metres per hour.
    /// </summary>
    public double DepthChangeRateMetresPerHour { get; set; }

    /// <summary>
    /// Gets or sets the pressure in pounds per square inch.
    /// </summary>
    public double PressurePsi { get; set; }

    /// <summary>
    /// Gets or sets the temperature in degrees Celsius.
    /// </summary>
    public double TemperatureCelsius { get; set; }

    /// <summary>
    /// Gets or sets the UTC timestamp of the reading.
    /// </summary>
    public DateTimeOffset TimestampUtc { get; set; }
}

File: src/DrillingTelemetry.Dashboard/src/app/features/dashboard/data-access/telemetry-reading.ts
import type { DrillingOperation } from './simulation-settings';

/** Describes how a telemetry reading was acquired. */
export enum TelemetryAcquisitionMode {
  RealTime = 'RealTime',
  HistoricalImport = 'HistoricalImport',
}

/** Persisted telemetry reading returned by the Processor API. */
export interface TelemetryReading {
  /** Identifier of the telemetry device. */
  readonly deviceId: string;

  /** Acquisition run that owns the sequence number. */
  readonly acquisitionSessionId: string;

  /** Sequence number assigned during acquisition. */
  readonly sequenceNumber: number;

  /** Identifier of the well being drilled. */
  readonly wellId: string;

  /** Name of the well being drilled. */
  readonly wellName: string;

  /** Identifier of the wellbore containing the tool. */
  readonly wellboreId: string;

  /** Name of the wellbore containing the tool. */
  readonly wellboreName: string;

  /** Distance travelled along the wellbore, in metres. */
  readonly measuredDepthMetres: number;

  /** Operation active when the reading was acquired. */
  readonly drillingOperation: DrillingOperation;

  /** Signed measured-depth change rate, in metres per hour. */
  readonly depthChangeRateMetresPerHour: number;

  /** Pressure in pounds per square inch. */
  readonly pressurePsi: number;

  /** Temperature in degrees Celsius. */
  readonly temperatureCelsius: number;

  /** Natural gamma ray measurement, in gAPI. */
  readonly gammaRayApi: number;

  /** UTC acquisition timestamp in ISO 8601 format. */
  readonly timestampUtc: string;

  /** Describes how the reading was acquired. */
  readonly acquisitionMode: TelemetryAcquisitionMode;
}
